driver-core/src/main/java/com/datastax/driver/core/tracing/TracingInfoFactory.java

```diff
@@ -0,0 +1,15 @@
+package com.datastax.driver.core.tracing;
+
+/** Factory of trace info objects. */
+public interface TracingInfoFactory {
+
+  /** Creates new trace info object, inheriting global context. */
+  TracingInfo buildTracingInfo();
+
+  /**
+   * Creates new trace info object, inheriting context from provided another trace info object.
+   *
+   * @param parent the trace info object to be set as the parent of newly created trace info object.
+   */
+  TracingInfo buildTracingInfo(TracingInfo parent);
```

Comment:
Will this variant ever be used? (that would mean a query execution within another query execution?)

Reply:
I think what we intended here is to enable inheriting Context from request span while creating execution span.